scripts/ci/openshift.sh

```diff
@@ -25,28 +26,43 @@ scale_worker_nodes() {
     if [[ -n "${ARTIFACT_DIR:-}" ]]; then
         local scale_debug_dir="${ARTIFACT_DIR}/openshift/scaling-debug"
         mkdir -p "${scale_debug_dir}"
-        oc -n openshift-machine-api get machineset -o json > "${scale_debug_dir}/machineset.json"
-        oc -n openshift-machine-api get machines -o json > "${scale_debug_dir}/machines.json"
+        oc -n openshift-machine-api get machineset -o json > "${scale_debug_dir}/initial_machineset.json"
+        oc -n openshift-machine-api get machines -o json > "${scale_debug_dir}/initial_machines.json"
     fi
 
-    original_count="$(get_running_worker_count)"
+    # note: timeout of this loop is handled by the calling context
+    while [[ "$(get_running_worker_count)" != "$expected_initial_worker_count" ]]; do
+        info "Initial machine state does not have the running worker count we expect ($expected_initial_worker_count)"
+        oc  -n openshift-machine-api get machines
+        oc -n openshift-machine-api get machineset -o json > "${scale_debug_dir}/polled_initial_machineset.json"
+        oc -n openshift-machine-api get machines -o json > "${scale_debug_dir}/polled_initial_machines.json"
+        sleep 20
+    done
+
     first_worker_machine_set="$(get_first_worker_machine_set)"
     current_replicas="$(oc -n openshift-machine-api get machineset "$first_worker_machine_set" -o json | jq -r '.spec.replicas')"
     new_machineset_count=$((current_replicas + increment))
     oc -n openshift-machine-api scale machineset "$first_worker_machine_set" --replicas="$new_machineset_count"
 
-    expected_count=$((original_count + increment))
+    expected_count=$((expected_initial_worker_count + increment))
     # note: timeout of this loop is handled by the calling context
     while [[ "$(get_running_worker_count)" != "$expected_count" ]]; do
         info "Current machine state does not have the running count we desire ($expected_count)"
```

Comment:
Here I'd also suggest to tweak the message
```suggestion
        info "Current machine state does not have the running count we desire ($expected_count). Waiting for machines to become running..."
```

---

Also, after looking at logs ([example](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/stackrox_stackrox/6002/pull-ci-stackrox-stackrox-master-ocp-4-10-qa-e2e-tests/1658159011482046464/artifacts/qa-e2e-tests/stackrox-e2e/build-log.txt)), I suggest logging this message after running `oc get` commands so that we first print the state and then tell what's wrong about it.

Reply:
:+1: